_posts/2022-02-04-rust-9.0.0.md

```diff
@@ -0,0 +1,141 @@
+cleaning up the last corner cases!
+
+Other notable recent additions to parquet are `UTF-8` validation on
+string data for improved security against malicious inputs.
+
+Planned upcoming work includes [pushing more
+filtering](https://github.com/apache/arrow-rs/issues/1191) directly
+into the parquet scan as well as an `async` writer.
+
+
+# `arrow`: performance, dyn kernels, and DecimalArray
+
+The [compute](https://docs.rs/arrow/latest/arrow/compute/index.html)
+kernels have been improved significantly in arrow `9.0.0`. Some [filter
+benchmarks](https://github.com/apache/arrow-rs/pull/1228#issue-1111889246)
+are twice as fast and the SIMD kernels are also [significantly
+faster](https://github.com/apache/arrow-rs/pull/1221). Many thanks to tustvold
+[tustvold](https://github.com/tustvold) and
+[jhorstmann](https://github.com/jhorstmann).
+[Additional substantial](https://github.com/apache/arrow-rs/pull/1248)
+improvements are likely to land in arrow `10.0.0`.
+
+We are working on new set of "dynamic" `dyn_` kernels (for example,
+[`eq_dyn`](https://docs.rs/arrow/8.0.0/arrow/compute/kernels/comparison/fn.eq_dyn.html))
+that make it easier to invoke the heavily optimized kernels provided
+by the `arrow` crate. Work is underway to expand the breadth of types
+supported by these new kernels to make them even more useful. Thanks
+to [matthewmturner](https://github.com/matthewmturner) and
+[viirya](https://github.com/viirya) for their help in this
+effort.
+
+While `arrow` has had basic support for `DecimalArray` since version
+`3.0.0`, support has been expanded for `Decimal` type in calculation
+kernels such as `sort`, `take` and `filter` thanks to some great
+contributions from [liukun4515](https://github.com/liukun4515). There
+is [ongoing work](https://github.com/apache/arrow-rs/pull/1223) to
+improve the API ergonomics and performance of `DecimalArray` as well.
+
+# Security
+
+The `6.4.0` release resolved the last outstanding
+[RUSTSEC](https://rustsec.org/)
+[advisory](https://github.com/rustsec/advisory-db/pull/1131) on the
+arrow crate and the `8.0.0` release resolved the last outstanding
+known security issues. While these security issues were mostly limited
+misuse of the low level "power user" APIs which most users do not (and
+should not) be using, it is good to tighten up that area.
+
+Now that `arrow-rs` is releasing major versions every other week, we
+are also able to update dependencies at the same pace, helping to
+ensure that security fixes upstream can flow more quickly to
+downstream projects.
+
+# Final shoutout
+It takes a community to build great software, and we would like to
+thank everyone who has contributed to the arrow-rs repository since
+the `7.0.0` release:
+
+```console
+git shortlog -sn 7.0.0..9.0.0
+    22  Raphael Taylor-Davies
+    18  Andrew Lamb
+     6  Helgi Kristvin Sigurbjarnarson
+     6  Remzi Yang
+     5  Jörn Horstmann
+     4  Liang-Chi Hsieh
+     3  Jiayu Liu
+     2  dependabot[bot]
```

Comment:
Truly a prolific individual :laughing:

Reply:
I figured I shouldn't discriminate against bots and give credit where it was due 😆